### PR TITLE
CET-370 C# stop using our fork of speakeasy's generation repo

### PR DESCRIPTION
**Pull request by rkeur7**
Link to ticket: https://shippo.atlassian.net/browse/CET-370

Validation Steps:
- Pull locally
- `make speakeasy-generate-sdk`
- `./.github/workflows/scripts/pre-publish-hook`
- Review Shippo/Shippo.csproj file for finalized configuration

### CODE REVIEW

File: .github/workflows/scripts/pre-publish-hook [hunk 1]
```diff
@@ -13,50 +13,6 @@ echo updating $f...
...
-        {
-          "+@Condition": "'"$condition"'",
-          "ContinuousIntegrationBuild": "true",
-          "EmbedUntrackedSources": "true",
-          "IncludeSymbols": "true",
-          "PublishRepositoryUrl": "true",
-          "SymbolPackageFormat": "snupkg"
-        },
-        {
-          "Copyright": "Copyright (c) Shippo 2024",
```
Comment (rkeur7) [inline .github/workflows/scripts/pre-publish-hook:28]: This is resolved by pulling from the `author` field in gen.yaml.
Result is in this format: `Copyright (c) <author> <current year>`

File: .github/workflows/scripts/pre-publish-hook [hunk 2]
```diff
@@ -13,50 +13,6 @@ echo updating $f...
...
-          "Description": ".NET library which integrates with Shippo Multi Carrier Shipping API. This library provides access to Shippo (goshippo.com) API capabilities. Including label generation, rating, tracking and more.",
```
Comment (rkeur7) [inline .github/workflows/scripts/pre-publish-hook:29]: This is resolved by info.description inside gen.yaml

File: .github/workflows/scripts/pre-publish-hook [hunk 3]
```diff
@@ -13,50 +13,6 @@ echo updating $f...
...
-          "PackageLicenseFile": "LICENSE",
-          "PackageIcon": "shippo.png",
```
Comment (rkeur7) [inline .github/workflows/scripts/pre-publish-hook:31]: This is automatically pulled from icon.png at root according to [docs](https://www.speakeasy.com/docs/publish-sdks#c-nuget)

Comment (rkeur7) [inline .github/workflows/scripts/pre-publish-hook:31]: These is automatically pulled from License & icon.png at root according to [docs](https://www.speakeasy.com/docs/publish-sdks#c-nuget)

File: .github/workflows/scripts/pre-publish-hook [hunk 4]
```diff
@@ -13,50 +13,6 @@ echo updating $f...
...
-          "PackageProjectUrl": "https://github.com/goshippo/shippo-csharp-sdk",
```
Comment (rkeur7) [inline .github/workflows/scripts/pre-publish-hook:32]: This is pulled from `externalDocs.url` inside `gen.yaml`

File: .github/workflows/scripts/pre-publish-hook [hunk 5]
```diff
@@ -13,50 +13,6 @@ echo updating $f...
...
-          "PackageTags": "USPS Fedex UPS API Shippo",
```
Comment (rkeur7) [inline .github/workflows/scripts/pre-publish-hook:33]: This is pulled from `csharp.packageTags` inside `gen.yaml`

File: .github/workflows/scripts/pre-publish-hook [hunk 6]
```diff
@@ -13,50 +13,6 @@ echo updating $f...
...
-          {
-            "+@Include": "..\LICENSE",
-            "+@Pack": "true",
-            "+@PackagePath": "\\"
-          },
-          {
-            "+@Include": "..\res\shippo.png",
-            "+@Pack": "true",
-            "+@PackagePath": "\\"
-          }
```
Comment (rkeur7) [inline .github/workflows/scripts/pre-publish-hook:51]: Resolved by License & logo comments above

File: .github/workflows/scripts/pre-publish-hook [hunk 7]
```diff
@@ -13,50 +13,6 @@ echo updating $f...
...
-  | .Project.PropertyGroup = [
-      .Project.PropertyGroup,
-      [
-        {
-          "+@Condition": "'"$condition"'",
-          "ContinuousIntegrationBuild": "true",
-          "EmbedUntrackedSources": "true",
-          "IncludeSymbols": "true",
-          "PublishRepositoryUrl": "true",
-          "SymbolPackageFormat": "snupkg"
```
Comment (rkeur7) [inline .github/workflows/scripts/pre-publish-hook:25]: Resolved by using setting `csharp.enableSourceLink` to `true` inside `gen.yaml`.
This requires further validation, might change behavior compared to other changes

File: .github/workflows/scripts/pre-publish-hook [hunk 8]
```diff
@@ -4,59 +4,16 @@ set -Eeuo pipefail
...
-        "PackageReference": {
-          "+@Include": "Microsoft.SourceLink.GitHub",
-          "+@PrivateAssets": "All",
-          "+@Version": "8.0.0"
-        }
-      }
-    ]
-  )' $f
+  )' $f &&
+sed -i "s/<EmbedAllSources>true<\/EmbedAllSources>/<EmbedUntrackedSources>true<\/EmbedUntrackedSources>/" $f
```
Comment (rkeur7) [inline .github/workflows/scripts/pre-publish-hook:16]: Convert `EmbedAllSources` to `EmbedUntrackedSources` to keep same amount of exposure (no clue why this was initially desired)

**Review by malcolmrebughini — APPROVED** [review]

**Pull request merged**